contracts/LifCrowdsale.sol

```diff
@@ -131,17 +150,23 @@ contract LifCrowdsale is Ownable, Pausable {
     // calculate token amount to be created
     uint256 tokens = weiAmount.mul(rate);
 
-    // update state
-    weiRaised = weiRaised.add(weiAmount);
-    purchases[beneficiary] = weiAmount;
-    tokensSold = tokensSold.add(tokens);
+    // check if we are in the public presale block range
+    if (rate == publicPresaleRate) {
```

Comment:
I think we should have separate functions for private presale, public presale, and ICO tokens purchases. Maybe we can keep a single getRate fn though... but the stages should be explicit when purchasing